src/app/inventory/note-hashtags.ts

```diff
@@ -4,41 +4,73 @@ import { maxBy } from 'es-toolkit';
 import { ItemInfos } from './dim-item-info';
 
 /**
- * Collects all hashtags from all item notes.
- *
- * Orders by use count, de-dupes case-insensitive, and picks the most popular capitalization.
+ * An object that can collect usage of hashtags and return their most popular form.
  */
-export function collectHashtagsFromInfos(itemInfos: ItemInfos) {
+export class HashTagTracker {
```

Comment:
This is one of the rare cases where I decided to use a class, because there's some state we want to keep track of as we go through the hashtags, and then you ask questions of that state.